lib/chef/resource/zypper_package.rb

```diff
@@ -0,0 +1,74 @@
+#
+# Author:: AJ Christensen (<aj@opscode.com>)
+# Copyright:: Copyright (c) 2008 Opscode, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/resource/package'
+require 'chef/provider/package/zypper'
+
+class Chef
+  class Resource
+    class ZypperPackage < Chef::Resource::Package
+
+      provides :zypper_package
+      provides :package, os: "linux", platform_family: [ "suse" ]
```

Comment:
according to lib/chef/provider/provider_mapping.rb, zypper is the default package provider for both "opensuse" and "suse".